### ISSUE TITLE
Standardize CLI documentation

### ISSUE THREAD

**Issue by bpmutter** [eslint/eslint#16475]
### ESLint version

v8.x

### What problem do you want to solve?

Different parts of the CLI have differing level of documentation. we should standardize the CLI commands reference docs for a clearer user experience.

### What do you think is the correct solution?

- Standardize reference documentation for [CLI](https://eslint.org/docs/latest/user-guide/command-line-interface).
- Have a unified and thorough way to document CLI arguments and examples.
- Do research on other popular CLI’s documentation to try to find models to base this documentation on.
- ~Investigate using an 11ty template to standardize each CLI option.~
   - Note: Investigated and I think this would add more complication than value.
   - Rather than standardization in a template, I think the best approach here is to just standardize the Markdown. Only if this becomes an issue going forward, should standardization template be considered.

### Participation

- [X] I am willing to submit a pull request for this change.

**Comment by bpmutter** [comment]
a convenient thing about ESLint the CLI only accepts flags. the only action is to lint. this makes standardizing the docs more straightforward. 

thoughts on how to structure docs for each flag:

````md
#### `--flag-name`

Short overview description.

* **Alias**: `-fn`
* **Argument Type**: Data type. Short description. If no data type, put 'No argument'
* **Multiple Arguments: Yes/No
* **Default Value**: whatever the default value is OR N/A

ADDITIONAL DESCRIPTION, AS NECESSARY.

##### `--flag-name` Example

PUT_WHATEVER_REQUIRED_FOR_EXAMPLE_HERE

````

**Comment by mdjermanovic** [comment]
> the only action is to lint

If one of these options is passed, there will be no linting:

* `--help` just shows help
* `--version` just shows the version.
* `--env-info` just shows the environment information.
* `--print-config` just shows the configuration for the given file.
* `--init` runs `npm init @eslint/config` to create a config file.
